### PR TITLE
Update spectrum handler to use new syntax

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR updates the spectrum handler to the new syntax.

### CODE REVIEW

File: skyportal/handlers/api/spectrum.py [hunk 1]
```diff
@@ -44,62 +43,67 @@
...
+    empty list.
+    """
+    if isinstance(str_list, str):
+        if len(str_list) == 0:
+            return []
+        elif "," in str_list:
+            return [c.strip() for c in str_list.split(",")]
         else:
-            raise TypeError('Must input a string!')
+            return [str_list.strip()]
```
Comment (guynir42) [inline skyportal/handlers/api/spectrum.py:101]: I think you get the same result by removing the last `else` and replacing the `elif` with an `else`.

File: skyportal/handlers/api/spectrum.py [hunk 2]
```diff
@@ -411,59 +415,62 @@ def get(self, spectrum_id=None):
...
+                ).first()
+                if spectrum is None:
+                    return self.error(
+                        f'Could not access spectrum {spectrum_id}.', status=403
+                    )
+                comments = (
+                    session.scalars(
+                        CommentOnSpectrum.select(
+                            session.user_or_token,
+                            options=[joinedload(CommentOnSpectrum.groups)],
```
Comment (guynir42) [inline skyportal/handlers/api/spectrum.py:432]: Do you have any idea why we join the groups here?

File: skyportal/handlers/api/spectrum.py [hunk 3]
```diff
@@ -1149,101 +1158,100 @@ def get(self, obj_id):
...
+                            "author": {
+                                **c.author.to_dict(),
+                                "gravatar_url": c.author.gravatar_url,
+                            },
+                        }
+                        for c in comments
+                    ),
+                    key=lambda x: x["created_at"],
+                    reverse=True,
                 )
```
Comment (guynir42) [inline skyportal/handlers/api/spectrum.py:1203]: As I mentioned in another PR, I'd prefer this as a loop.

**Review by guynir42 — APPROVED** [review]

File: skyportal/handlers/api/spectrum.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/spectrum.py:101]: I had to stare at this comment for a while to try to figure out why it was the case ;)

File: skyportal/handlers/api/spectrum.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/spectrum.py:432]: Nope.

File: skyportal/handlers/api/spectrum.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/handlers/api/spectrum.py:1203]: I think this is another PR if we go that route.

**Pull request merged**